node/impl/full/eth_trace.go

```diff
@@ -0,0 +1,351 @@
+		LastByteCode: nil,
+	}
+
+	trace.SetCallType("call")
+
+	if et.Msg.Method == builtin.MethodsEVM.InvokeContract {
+		log.Debugf("COND1 found InvokeContract call at height: %d", height)
+
+		// TODO: ignore return errors since actors can send gibberish and we don't want
+		// to fail the whole trace in that case
+		trace.Action.Input, err = decodePayload(et.Msg.Params, et.Msg.ParamsCodec)
+		if err != nil {
+			return xerrors.Errorf("buildTraces: %w", err)
+		}
+		trace.Result.Output, err = decodePayload(et.MsgRct.Return, et.MsgRct.ReturnCodec)
+		if err != nil {
+			return xerrors.Errorf("buildTraces: %w", err)
+		}
+	} else if et.Msg.To == builtin.EthereumAddressManagerActorAddr &&
+		et.Msg.Method == builtin.MethodsEAM.CreateExternal {
+		log.Debugf("COND2 found CreateExternal call at height: %d", height)
+		trace.Action.Input, err = decodePayload(et.Msg.Params, et.Msg.ParamsCodec)
+		if err != nil {
+			return xerrors.Errorf("buildTraces: %w", err)
+		}
+
+		if et.MsgRct.ExitCode.IsSuccess() {
+			// ignore return value
+			trace.Result.Output = nil
+		} else {
+			// return value is the error message
+			trace.Result.Output, err = decodePayload(et.MsgRct.Return, et.MsgRct.ReturnCodec)
+			if err != nil {
+				return xerrors.Errorf("buildTraces: %w", err)
+			}
+		}
+
+		// treat this as a contract creation
+		trace.SetCallType("create")
+	} else {
+		// we are going to assume a native method, but we may change it in one of the edge cases below
+		// TODO: only do this if we know it's a native method (optimization)
+		trace.Action.Input, err = handleFilecoinMethodInput(et.Msg.Method, et.Msg.ParamsCodec, et.Msg.Params)
+		if err != nil {
+			return xerrors.Errorf("buildTraces: %w", err)
+		}
+		trace.Result.Output, err = handleFilecoinMethodOutput(et.MsgRct.ExitCode, et.MsgRct.ReturnCodec, et.MsgRct.Return)
+		if err != nil {
+			return xerrors.Errorf("buildTraces: %w", err)
+		}
+	}
+
+	// TODO: is it OK to check this here or is this only specific to certain edge case (evm to evm)?
+	if et.Msg.ReadOnly {
+		trace.SetCallType("staticcall")
+	}
+
+	// there are several edge cases that require special handling when displaying the traces. Note that while iterating over
+	// the traces we update the trace backwards (through the parent pointer)
+	if parent != nil {
+		// Handle Native actor creation
+		//
+		// Actor A calls to the init actor on method 2 and The init actor creates the target actor B then calls it on method 1
+		if parent.Action.FilecoinTo == builtin.InitActorAddr &&
+			parent.Action.FilecoinMethod == builtin.MethodsInit.Exec &&
+			et.Msg.Method == builtin.MethodConstructor {
+			log.Debugf("COND3 Native actor creation! method:%d, code:%s, height:%d", et.Msg.Method, et.Msg.CodeCid.String(), height)
+			parent.SetCallType("create")
+			parent.Action.To = to
+			parent.Action.Input = []byte{0xFE}
+			parent.Result.Output = nil
+
+			// there should never be any subcalls when creating a native actor
+			return nil
```

Comment:
This isn't correct but no native actor _currently_ calls another in the constructor. Can we change this comment? Or just change the code, supporting this use-case shouldn't be complicated unless I'm mistaken.